File: Projects/Enhancement-2/Original Files/AppointmentService.java
// Author Name: Owen Lovell

// Date: 10/06/2024

// Course ID: CS-320-13376-M01

// Description: This is the appointment service class that adds or deletes an appointment

package module5;

import java.util.ArrayList;

public class AppointmentService {

	ArrayList<Appointment> appointmentList = new ArrayList<Appointment>();
	// Add new Appointment
	public void addAppointment(String name, String description) {
		Appointment appt = new Appointment(name, description);
		appointmentList.add(appt);
	}
		
	// delete appointment
	public void deleteAppointment(String appointmentID) {
		appointmentList.remove(getAppointment(appointmentID));
	}
	
	public Appointment getAppointment(String appointmentID) {
		return appointmentList.stream()
			.filter(appointment -> appointment.getAppointmentID().equals(appointmentID))
			.findFirst()
			.orElseThrow(() -> new RuntimeException("Couldn't find Appointment with that AppointmentID"));
		
	}
	
}

File: Projects/Enhancement-1/Enhanced/appointment_service.py
# Author: Owen Lovell
# Date Updated: 12/8/2025
# Course ID: CS-499-17242-M01

"""
Description: The Appointment Service is used to create an appointment object, the object is then placed in a list
"""


from appointment import *

appointment_list = []


class AppointmentList:

    def add_appointment(self, name, description):
        new_appointment = Appointment(name, description)  # This is why 'AppointmentID is in list
        appointment_list.append(new_appointment)

    # this removes the first instance of name in the list
    def remove_appointment(self, name):
        appointment_list.remove(name)

    # get specific appointment for testing etc
    def get_appointment(self, name):
        try:
            index = appointment_list.index(name)
            print("Found at index: ")
            print(index)
            return index
        except ValueError:
            print("Appointment Not Found")

    # function to print the list to a string
    def print_appointments(self):
        print('\n'.join(map(str, appointment_list)))
